artemis/api.py

```diff
@@ -84,12 +96,32 @@ def add(
     elif not disabled_modules:
         disabled_modules = Config.Miscellaneous.MODULES_DISABLED_BY_DEFAULT
 
+    # Validate module configurations if provided
+    if module_configs:
+        for module_name, config in module_configs.items():
+            config_class = ConfigurationRegistry().get_configuration_class(module_name)
+            if config_class:
+                try:
+                    config_instance = config_class.deserialize(config)
+                    if not config_instance.validate():
+                        raise ValueError(f"Invalid configuration for module {module_name}")
+                except ValueError as e:
+                    raise HTTPException(status_code=400, detail=f"Invalid configuration for {module_name}: {str(e)}")
+    # If no module_configs provided, get configs from Redis as fallback
```

Comment:
I don't think this fallback is particularily needed, as it adds a dependency on the global state